www/widgets/rt.js

```diff
@@ -164,55 +166,90 @@ class ssl {
 }
 
 class tkinter { 
-    static Tk = wrap_class(class {
-        constructor() {
-            this.elt = rt_constants.ROOT_CANVAS;
+    static Tk = function(...args) {
+        if (rt_constants.WINDOW) {
+            console.log('reused')
+            return rt_constants.WINDOW;
         }
 
-        bind(key, fn) {
-            if (['<Up>', '<Down>', '<Left>', '<Right>'].indexOf(key) !== -1) {
-                window.addEventListener("keydown", function(e) {
-                    if (e.key == 'Arrow' + key.substr(1, key.length-2)) {
-                        e.preventDefault();
-                        fn({}).catch(on_error);
-                    }
-                });
-            } else if (key == '<Return>') {
-                window.addEventListener("keydown", function(e) {
-                    if (e.key == 'Enter') {
-                        e.preventDefault();
-                        fn({}).catch(on_error);
-                    }
-                });
-            } else if (['<Button-1>', '<Button-2>', '<Button-3>'].indexOf(key) !== -1) {
-                this.elt.addEventListener("mousedown", function(e) {
-                    if (e.button == key.substr(8, 1) - 1) {
-                        fn({ x: e.offsetX, y: e.offsetY }).catch(on_error);
-                    }
-                });
-            } else if (key == '<Configure>') {
-                window.addEventListener("resize", function(e) {
-                    fn({}).catch(on_error);
-                });
-            } else if (key == '<Key>') {
-                window.addEventListener("keydown", function(e) {
-                    if (e.key.length == 1) {
-                        e.preventDefault();
-                        fn({ char: e.key }).catch(on_error);
+        return wrap_class(class {
```

Comment:
I'm a little confused what was changed by the diff here.

Reply:
The Tk function now does this:

If `rt_constants.WINDOW` doesn't exist, make it in a wrap_class and return it. Otherwise return `rt_constants.WINDOW`. This implements a singleton pattern, which allows us to preserve the Tk object across resizes, and rebind it to new callback functions when the `Browser` object is re-created.